de.metas.swat/de.metas.swat.base/src/main/java/de/metas/order/impl/OrderLineBL.java

```diff
@@ -956,30 +961,69 @@ public void copyOrderLineCounter(final org.compiere.model.I_C_OrderLine line, fi
 	public void updateNoPriceConditionsColor(final I_C_OrderLine orderLine)
 	{
 
-		final String colorName = Services.get(ISysConfigBL.class).getValue(SYSCONFIG_NoPriceConditionsColorName, "-");
+		final String colorName = getNoPricingConditionsColorName();
 
 		final int discountSchemaBreakId = orderLine.getM_DiscountSchemaBreak_ID();
 
 		if (discountSchemaBreakId > 0)
 		{
 			// the discountSchemaBreak was eventually set. The color warning is no longer needed
 			orderLine.setNoPriceConditionsColor_ID(-1);
-			
+
 			return;
 		}
 
 		final int colorId = getNoPriceConditionsColorId(colorName);
-		
+
 		if (colorId > 0)
 		{
 			orderLine.setNoPriceConditionsColor_ID(colorId);
 		}
 
 	}
 
+	private String getNoPricingConditionsColorName()
+	{
+		return Services.get(ISysConfigBL.class).getValue(SYSCONFIG_NoPriceConditionsColorName, "-");
+	}
+
 	private int getNoPriceConditionsColorId(final String name)
 	{
 		return Services.get(IColorRepository.class).getColorIdByName(name);
 	}
 
+	@Override
+	public void failForMissingPricingConditions(final de.metas.adempiere.model.I_C_Order order)
+	{
+		final boolean mandatoryPricingConditions = isMandatoryPricingConditions();
+
+		if (!mandatoryPricingConditions)
+		{
+			// nothing to do
+			return;
+		}
+
+		final List<I_C_OrderLine> orderLines = Services.get(IOrderDAO.class).retrieveOrderLines(order);
+
+		final boolean existsOrderLineWithNoPricingConditions = orderLines
+				.stream()
+				.anyMatch(orderLine -> orderLine.getM_DiscountSchemaBreak_ID() <= 0);
```

Comment:
refactor `orderLine.getM_DiscountSchemaBreak_ID() <= 0` and introduce hasNoPricingConditions(orderLine).

Reply:
in next tasks that conditions will be a bit more complicated.